celery/backends/base.py

```diff
@@ -454,7 +466,10 @@ def store_result(self, task_id, result, state,
                             self.max_sleep_between_retries_ms, True) / 1000
                         self._sleep(sleep_amount)
                     else:
-                        raise_with_traceback(BackendStoreError("failed to store result on the backend", task_id=task_id, state=state))
+                        _raise_with_traceback(
+                            BackendStoreError("failed to store result on the backend", task_id=task_id, state=state),
+                            traceback,
```

Comment:
@mchataigner is this the traceback you intend to raise with?

If you pass `None` in here, I think it's the same as the plain `raise BackendStoreError(...)`